src/components/SqlEditor.tsx

```diff
@@ -1,44 +1,55 @@
 import React, { useState } from 'react';
-import { QueryEditorProps } from '@grafana/data';
+import { CoreApp, QueryEditorProps } from '@grafana/data';
 import { CodeEditor } from '@grafana/ui';
-import { Datasource } from '../data/CHDatasource';
+import { Datasource } from 'data/CHDatasource';
 import { registerSQL, Range, Fetcher } from './sqlProvider';
-import { CHQuery, CHConfig, QueryType, CHSQLQuery } from '../types';
-import { styles } from '../styles';
+import { CHConfig } from 'types/config';
+import { CHQuery, EditorType, CHSqlQuery } from 'types/sql';
+import { styles } from 'styles';
 import { fetchSuggestions as sugg, Schema } from './suggestions';
 import { selectors } from 'selectors';
-import { getFormat } from './editor';
 import { validate } from 'data/validate';
+import { mapQueryTypeToGrafanaFormat } from 'data/utils';
+import { QueryType } from 'types/queryBuilder';
+import { QueryTypeSwitcher } from 'components/queryBuilder/QueryTypeSwitcher';
 
-type SQLEditorProps = QueryEditorProps<Datasource, CHQuery, CHConfig>;
+type SqlEditorProps = QueryEditorProps<Datasource, CHQuery, CHConfig>;
 
 interface Expand {
   height: string;
   icon: 'plus' | 'minus';
   on: boolean;
 }
 
-export const SQLEditor = (props: SQLEditorProps) => {
+export const SqlEditor = (props: SqlEditorProps) => {
   const defaultHeight = '150px';
-  const { query, onRunQuery, onChange, datasource } = props;
+  const { app, query, onChange, datasource } = props;
+  const sqlQuery = query as CHSqlQuery;
```

Comment:
Can we declare `query` type as a union and enforce it via contract, instead of casting manually?